dali/c_api/c_api.h

```diff
@@ -69,6 +75,44 @@ extern "C" {
       int cpu_prefetch_queue_depth,
       int gpu_prefetch_queue_depth);
 
+  /**
+   * @brief Feed the data to ExternalSource as contiguous memory.
+   *
+   * @param pipe_handle Pointer to pipeline handle
+   * @param name Pointer to a null terminated string with the name of the External Source to be fed
+   * @param device Device of the supplied memory. Only CPU is supported.
+   * @param data_ptr Pointer to contiguous buffer containing all samples
+   * @param data_type Type of the provided data
+   * @param shape Pointer to an array containing shape of all samples concatenated one after
+   *              another. Should contain batch_size * sample_dim elements.
+   * @param sample_dim The dimensionality of the data.
+   * @param layout_str Optional layout provided as a pointer null terminated string.
+   *                   Can be set to NULL.
```

Comment:
Maybe setting it to `NULL` as default argument might be a good idea?

Reply:
If we want to stick with this being C API (at least pretend that it is such), there is no such thing as default value in C.